fabtests/common/hmem_neuron.c

```diff
@@ -0,0 +1,360 @@
+/*
+ * (C) Copyright 2020 Hewlett Packard Enterprise Development LP
+ * Copyright (c) 2021 Amazon.com, Inc. or its affiliates.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "hmem.h"
+#include "shared.h"
+
+#ifdef HAVE_NEURON
+
+#include <dlfcn.h>
+#include <ft_list.h>
+
+#include "nrt/nrt.h"
+#include "nrt/nrt_experimental.h"
+
+/* Size of temporary buffer to alloc for memset */
+#define NEURON_MEMSET_BUF_SIZE 65536
+
+struct neuron_ops {
+	NRT_STATUS (*nrt_tensor_allocate)(nrt_tensor_placement_t tensor_placement,
+					  int logical_nc_id, size_t size,
+					  const char *name, nrt_tensor_t **tensor);
+	void (*nrt_tensor_free)(nrt_tensor_t **tensor);
+	void *(*nrt_tensor_get_va)(const nrt_tensor_t *tensor);
+	NRT_STATUS (*nrt_tensor_read)(const nrt_tensor_t *tensor, void *buf, size_t offset, size_t size);
+	NRT_STATUS (*nrt_tensor_write)(nrt_tensor_t *tensor, const void *buf, size_t offset, size_t size);
+	NRT_STATUS (*nrt_init)(nrt_framework_type_t framework, const char *fw_version, const char *fal_version);
+};
+
+static void *neuron_handle = NULL;
+static struct neuron_ops neuron_ops;
+
+/*
+ * List to lookup the handle based on the pointer. Not optimal, but probably
+ * fine for fabtests and this is better than changing the alloc/free functions
+ * to pass a handle and pointer.
+ */
+struct nrt_allocation {
```

Comment:
If nrt is used as the prefix by nrt.h and that package, maybe we should stick with the neuron prefix everywhere in OFI.

Reply:
Good point, fixed.